src/daemon/keymap.c

```diff
@@ -829,6 +861,87 @@ void process_input_urb(void* context, unsigned char* buffer, int urblen, ushort
                                 if(matched)
                                     CLEAR_KEYBIT(targetkb->input.keys, 122);
                             }
+                        }else if(IS_K70CORERGB(targetkb)){
+                            #define K70_CORE_MEDIA_MASK 0x60
+                            if(buffer[1] == 0x5 && buffer[2] == K70_CORE_MEDIA_MASK) {
+                                // This is a packet from the volume knob
+                                // We need to handle it appropriately then clear the keys so we stop getting garbage
+                                if(buffer[4] == 0x01) {
+                                    // Volume knob turned right
+                                    // Store
+                                    uchar media_key_state_1 = targetkb->input.keys[12];
+                                    uchar media_key_state_2 = targetkb->input.keys[16];
+
+                                    // Apply fresh key data
+                                    memset(targetkb->input.keys, 0, sizeof targetkb->input.keys);
+                                    SET_KEYBIT(targetkb->input.keys, 103); // volup
+
+                                    // Restore
+                                    targetkb->input.keys[12] = (targetkb->input.keys[12] & ~K70_CORE_MEDIA_MASK) | (media_key_state_1 & K70_CORE_MEDIA_MASK);
+                                    //targetkb->input.keys[16] = (targetkb->input.keys[16] & ~NXP_STRAFE_VOL_MASK) | (media_key_state_2 & NXP_STRAFE_VOL_MASK);
+                                    // We can just copy this byte as-is, as there are no M or G keys in the strafe.
+                                    targetkb->input.keys[16] = media_key_state_2;
+                                } else if(buffer[4] == 0xFF) {
+                                    // Volume knob turned left
+                                    // Store
+                                    uchar media_key_state_1 = targetkb->input.keys[12];
+                                    uchar media_key_state_2 = targetkb->input.keys[16];
+
+                                    // Apply fresh key data
+                                    memset(targetkb->input.keys, 0, sizeof targetkb->input.keys);
```

Comment:
Zeroing out the keys will let go of any keys that were held down.

Reply:
This is the thing that fixed the media dial sending garbled text and breaking the keyboard. I'll see if I can avoid clearing it but I think this is the thing that made it work. It is possible that excluding it from DEV_HAS_VOLWHEEL per your recommendation on IRC chat made this unnecessary. Note: This is only for the media dial's volume up/down which shouldn't be used with held keys.